### PR TITLE
enrich http_context if pre-existing

### PR DESCRIPTION
**Pull request by borft**
In this PR:
- enrich pre-existing HttpContext with `remote_address` if present, otherwise, behaviour is the same as it was
-  add some tests around the HttpContext
- fix typo in `referer` to `referrer` in the schema

### CODE REVIEW

**Review by thijs-obj — COMMENTED** [review]
Some small comments

File: backend/objectiv_backend/end_points/collector.py [hunk 1]
```diff
@@ -165,45 +163,61 @@ def set_time_in_events(events: EventDataList, current_millis: int, client_millis
...
+        remote_address = request.headers['X-Real-IP']
+    elif 'X-Forwarded-For' in request.headers:
         # x-forwarded-for headers take the form of:
         # client proxy_1...proxy_n
         # we're interested in proxy_n, which is the last node that connects to us, and as such
         # has to be an Internet routed proxy. proxies before it, including the client may be internal
         # and non-routable addresses.
-        hosts = str(flask.request.headers['X-Forwarded-For']).split()
+        hosts = str(request.headers['X-Forwarded-For']).split()
         if len(hosts) > 1:
```
[the remark this answers is not in the document]
Comment (borft) [reply backend/objectiv_backend/end_points/collector.py:190]: good point

File: backend/tests/schema/test_schema.py [hunk 2]
```diff
@@ -151,3 +163,72 @@ def test_add_context_to_incorrect_scope():
 
     # check if event is not valid anymore
     assert(validate_event_adheres_to_schema(event_schema=event_schema, event=event) != [])
+
+
+def _get_http_context():
```
[the remark this answers is not in the document]
Comment (borft) [reply backend/tests/schema/test_schema.py]: I see your point, fixed

**Review by thijs-obj — APPROVED** [review]
lgtm

**Pull request merged**